portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/CustomEventsEventAnalysis.testcase

```diff
@@ -178,6 +178,108 @@ definition {
+	@priority = "3"
+	test CanPaginationEventAnalysisResult {
+		var defaultEventName = "pageViewed";
+		var attributeName = "pageTitle";
+		var layoutNameList = "Custom 1,Custom 2,Custom 3,Custom 4,Custom 5";
+		var firstPageList = "ac page,custom 2,custom 3,custom 4,custom 5";
+		var secondPageList = "custom 1";
+		var dataSource = StringUtil.extractFirst("${assignedPropertyName}", " ");
+
+		task ("Sign in as new user") {
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "test@liferay.com",
+				userLoginFullName = "test test");
+		}
+
+		task ("Create new pages and put the web content in the pages") {
+			for (var layoutName : list "${layoutNameList}") {
+				ACUtils.addPage(layoutName = "${layoutName}");
+
+				ACCustomEvents.addWCWithCustomEventToPage(layoutName = "${layoutName}");
+			}
+		}
+
+		task ("View the pages") {
+			for (var pageName : list "AC Page,${layoutNameList}") {
+				ACUtils.navigateToSitePage(
+					actionType = "View WC",
+					pageName = "${pageName}",
+					siteName = "Site Name");
+			}
+		}
+
+		task ("Close sessions") {
+			ACUtils.closeAllSessionsAndWait();
+		}
+
+		task ("Switch the property in AC and go to event analysis") {
+			ACUtils.launchAC();
+
+			ACProperties.switchProperty(propertyName = "${assignedPropertyName}");
+
+			ACNavigation.goToEventAnalysis();
+		}
+
+		task ("Add the default event to analysis") {
+			ACEventAnalysis.addEvent(customEventName = "${defaultEventName}");
+		}
+
+		task ("Add the breakdown to analysis") {
+			ACEventAnalysis.addBreakdown(attributeNameList = "${attributeName}");
+		}
+
+		task ("Change the time filter to 24 hours") {
+			ACTimeFilter.clickTimeFilterButton();
+
+			ACTimeFilter.setLast24Hours();
+		}
+
+		task ("Check pagination for "20" items filter") {
+			ACUtils.viewResultsMessage(results = "Showing 1 to 6 of 6 entries.");
+
+			ACUtils.viewResultsPages(pageNumber = "1");
+		}
+
+		task ("Change the pagination to "5" items") {
+			ACUtils.changePagination(itemsPerPage = "5");
+
+			ACUtils.viewResultsMessage(results = "Showing 1 to 5 of 6 entries.");
+
+			ACUtils.viewResultsPages(pageNumber = "1");
+
+			ACUtils.viewResultsPages(pageNumber = "2");
+		}
+
+		task ("Check the pages that appear in the pagination") {
+			for (var pageName : list "${firstPageList}") {
+				ACEventAnalysis.viewAnalysisInformation(informationValueList = "${pageName} - site name - ${dataSource}");
+			}
+		}
+
+		task ("Check that the page on the second page does not appear on the first") {
+			ACEventAnalysis.viewAnalysisInformationNotPresent(informationValueList = "${secondPageList} - site name - ${dataSource}");
+		}
+
+		task ("Navigate to the next page") {
+			ACUtils.checkPreviousNextButton(nextButton = "true");
+
+			ACUtils.viewResultsMessage(results = "Showing 6 to 6 of 6 entries.");
+		}
+
+		task ("Check the page that appear in the pagination") {
+			ACEventAnalysis.viewAnalysisInformation(informationValueList = "${secondPageList} - site name - ${dataSource}");
+		}
```

Comment:
Hey @ViniLopes20 nice PR. I just didn't understand something, when you go to the second page you are looking for `custom 1`, in AC when we create an analysis it's not in alphabetical order?

Reply:
Hey @manuferreira, the ordering in the analysis result is by the number of iterations of the event and all of them except `ac page` is the same value (value equal to 3). So sorting is done differently because of that

Reply:
nice, thank you Vini.